layers/cmd/ingest_csv.go

```diff
@@ -152,11 +151,12 @@ func (ci *CSVIngester) Run(pipeline *pipeline.PipelineContext) error {
 					doneErr = err
 					return
 				}
-				_, err = ls.Ingest(builder, parsed)
+				r, err := ls.Ingest(builder, parsed)
 				if err != nil {
 					doneErr = err
 					return
 				}
+				r.SetProperty("Filename", entryInfo.GetName())
```

Comment:
Not "fileName", use what you got from Config.SourceProperty. If empty, use "source"